source/faq.txt

```diff
@@ -148,6 +148,32 @@ When ``MongoClient.close()`` is called by any thread, the driver
 closes all idle sockets and closes all sockets that are in
 use as they are returned to the pool.
 
+How do I prevent the "IllegalStateException: state should be: open" error?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You may encounter this exception if you call an operation on a ``MongoClient``
+instance that closed its connections to MongoDB. The client's connections
+can be closed for one of the following reasons:
+
+- The ``close()`` method was called on that ``MongoClient`` instance prior
+  to the operation call.
+- Another thread or process closed the connection. This can happen when
```

Comment:
I don't understand what the difference is between the first and second bullet.  The only way to "close the connection" is to call the close method on the MongoClient.  The difference only seems to be where the close is happening (how close/far it is from the code that is getting the exception, whether in the same thread or a different thread.  

Also, it's not possible for another process to directly close a MongoClient.  It would have to happen indirectly through communication through a socket, or some signal, etc.

Also, it may be the

Reply:
I see. I assumed that [this response](https://www.mongodb.com/community/forums/t/illegalstateexception-state-should-be-open/223912/4) was referring to an executor service operating on a pool of threads.  Would you prefer I remove this list item to avoid any confusion? Is there any info you'd like to add? Thanks!

Reply:
Yes, please remove.  You could point out that often MongoClient instances are singletons that are accessible by multiple threads, and so the code which closes the MongoClient may be distant from the code receiving the exception, perhaps in another thread of execution.